ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -135,197 +98,271 @@ export default class ConfirmDecryptMessage extends Component {
+  const notice = t('decryptMessageNotice', [txData.msgParams.origin]);
+
+  return (
+    <div className="request-decrypt-message__container">
+      <Header />
       <div className="request-decrypt-message__body">
-        {this.renderAccountInfo()}
-        <div className="request-decrypt-message__visual">
-          <section>
-            {targetSubjectMetadata?.iconUrl ? (
-              <img
-                className="request-decrypt-message__visual-identicon"
-                src={targetSubjectMetadata.iconUrl}
-                alt=""
-              />
-            ) : (
-              <i className="request-decrypt-message__visual-identicon--default">
-                {name.charAt(0).toUpperCase()}
-              </i>
-            )}
-            <div className="request-decrypt-message__notice">{notice}</div>
-          </section>
-        </div>
-        <div className="request-decrypt-message__message">
-          <div className="request-decrypt-message__message-text">
-            {!hasDecrypted && !hasError ? txData.msgParams.data : rawMessage}
-            {hasError ? errorMessage : ''}
-          </div>
-          <div
-            className={classnames('request-decrypt-message__message-cover', {
-              'request-decrypt-message__message-lock--pressed':
-                hasDecrypted || hasError,
-            })}
-          />
-          <div
-            className={classnames('request-decrypt-message__message-lock', {
-              'request-decrypt-message__message-lock--pressed':
-                hasDecrypted || hasError,
-            })}
-            onClick={(event) => {
-              decryptMessageInline(txData, event).then((result) => {
-                if (result.error) {
-                  this.setState({
-                    hasError: true,
-                    errorMessage: this.context.t('decryptInlineError', [
-                      result.error,
-                    ]),
-                  });
-                } else {
-                  this.setState({
-                    hasDecrypted: true,
-                    rawMessage: result.rawSig,
-                  });
-                }
-              });
-            }}
-          >
-            <div className="request-decrypt-message__message-lock__container">
-              <i className="fa fa-lock fa-lg request-decrypt-message__message-lock__container__icon" />
-              <div className="request-decrypt-message__message-lock__container__text">
-                {t('decryptMetamask')}
-              </div>
+        <Account
+          conversionRate={conversionRate}
+          currentCurrency={currentCurrency}
+          fromAccount={fromAccount}
+          nativeCurrency={nativeCurrency}
+          requesterAddress={requesterAddress}
+        />
+        <VisualSection
+          name={name}
+          notice={notice}
+          targetSubjectMetadata={targetSubjectMetadata}
+        />
+        <div className="request-decrypt-message__message-container">
```

Comment:
Is it worth extracting this into a local `MessageBody` component?

Reply:
Done